x-pack/plugins/observability/public/pages/slo_details/components/slo_details.tsx

```diff
@@ -5,14 +5,49 @@
  * 2.0.
  */
 
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
 import { SLOWithSummaryResponse } from '@kbn/slo-schema';
 import React from 'react';
 
+import { useFetchHistoricalSummary } from '../../../hooks/slo/use_fetch_historical_summary';
+import { ErrorBudgetChartPanel } from './error_budget_chart_panel';
+import { Overview as Overview } from './overview';
+import { SliChartPanel } from './sli_chart_panel';
+
 export interface Props {
   slo: SLOWithSummaryResponse;
 }
 
-export function SloDetails(props: Props) {
-  const { slo } = props;
-  return <pre data-test-subj="sloDetails">{JSON.stringify(slo, null, 2)}</pre>;
+export function SloDetails({ slo }: Props) {
+  const { isLoading: historicalSummaryLoading, sloHistoricalSummaryResponse } =
+    useFetchHistoricalSummary({ sloIds: [slo.id] });
+
+  const errorBudgetBurnDownData = (sloHistoricalSummaryResponse[slo.id] ?? []).map((data) => ({
```

Comment:
Is this processing only useful on this page or could it also be used within the graph on an SLO Item? If so, perhaps we could take this processing inside the useFetchHistoricalSummary hook?

Reply:
This hook is used in both the details and listing page. I like the hook to fetch the response as it is, because it groups per slo, different data, e.g. `Record<sloId, DATA>`.
But agree, the formatting logic to use in the graph could be extracted into a common function. I don't think we need a hook for that, do you?

Reply:
@CoenWarmer Here's one solution:

The common function would be something like:
```
type DataType = 'error_budget_remaining' | 'error_budget_consumed' | 'sli_value';

function formatHistoricalData(
  historicalSummary: FetchHistoricalSummaryResponse[string] | undefined,
  dataType: DataType
): Data[] {
  function getDataValue(data: FetchHistoricalSummaryResponse[string][number]) {
    switch (dataType) {
      case 'error_budget_consumed':
        return data.errorBudget.consumed;
      case 'error_budget_remaining':
        return data.errorBudget.remaining;
      default:
        return data.sliValue;
    }
  }

  if (!historicalSummary) {
    return [];
  }

  return historicalSummary.map((data) => ({
    key: new Date(data.date).getTime(),
    value: data.status === 'NO_DATA' ? undefined : getDataValue(data),
  }));
}
```


Then the usage becomes:
```
const { isLoading: historicalSummaryLoading, sloHistoricalSummaryResponse = {} } =
    useFetchHistoricalSummary({ sloIds: [slo.id] });

  const errorBudgetBurnDownData = formatHistoricalData(
    sloHistoricalSummaryResponse[slo.id],
    'error_budget_remaining'
  );
  const historicalSliData = formatHistoricalData(sloHistoricalSummaryResponse[slo.id], 'sli_value');
```


I wonder if that's not over-engineered vs the two inlined map. What do you think?

Reply:
Yeah, if this is only used in this case I would say leave it as is. If the same logic is needed elsewhere it can always be extracted to this function later.